src/riak_kv_vnode.erl

```diff
@@ -813,7 +817,33 @@ handle_command({get_index_entries, Opts},
         false ->
             lager:error("Backend ~p does not support incorrect index query", [Mod]),
             {reply, ignore, State}
-    end.
+    end;
+
+handle_command(?KV_W1C_PUT_REQ{bkey={Bucket, Key}, encoded_obj=EncodedVal, type=Type},
```

Comment:
Some general comments, and you may leave them for later refactoring/TODO:

1. I really dislike using `case` where a function clause is possible. The nested cases *look* similar but aren't completely, and the nesting makes that harder to see.
2. You use manual `reply` when that could instead be included in the return value of each clause. I think this obfuscates the code, especially when you discard the value of `Reply` in the first branch.
3. The return values of the branches have different types and you don't match on them. Dialyzer will complain about this. Furthermore, you export `ModState2` from the case when you could just return it.